### PR TITLE
test(governance): governance tests refactor

### LINKED ISSUE #2950: Convert Governance tests to typescript
## The Chore

In keeping with a lot of the other tests. The Governance tests should be changed to typescript for a bit more robustness

## Tasks

- [x] Refactor existing staking and association tests - there's a lot of repeated code in those tests that can be refactored. 
- [ ] Move test files to typescript and resolve any errors
- [ ] Change Cypress commands which are not in libs to functions

### PR DESCRIPTION
**Pull request by jtsang586**
# Related issues 🔗

Closes #2950 

# Description ℹ️

- Refactors staking flow tests by reducing the amount of repeated code
- Convert all javascript test files to Typescript
- Update common functions to work with typescript
- Test fixes

### CODE REVIEW

File: apps/governance-e2e/src/integration/view/validators.cy.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+const validatorId = '[data-testid="validator-id"]';
+const validatorPubKey = '[data-testid="validator-public-key"]';
+const ethAddressLink = '[data-testid="link"]';
+const validatorStatus = '[data-testid="validator-status"]';
+const totalStake = '[data-testid="total-stake"]';
+const pendingStake = '[data-testid="pending-stake"]';
+const stakedByOperator = '[data-testid="staked-by-operator"]';
+const stakedByDelegates = '[data-testid="staked-by-delegates"]';
+const stakeShare = '[data-testid="stake-percentage"]';
+const stakedByOperatorToolTip = "[data-testid='staked-operator-tooltip']";
```
Comment (sam-keen) [inline apps/governance-e2e/src/integration/view/validators.cy.ts:22]: Nitpick: inconsistent use of single and double quotes here

File: apps/governance-e2e/src/support/common.functions.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  validators = '[href="/validators"]',
+  rewards = '[href="/rewards"]',
+  withdraw = '[href="/token/withdraw"]',
+  proposals = '[href="/proposals"]',
+  pageSpinner = '[data-testid="splash-loader"]',
+  supply = '[href="/token/tranches"]',
+  token = '[href="/token"]',
+}
+
+export function convertTokenValueToNumber(subject: string) {
```
Comment (sam-keen) [inline apps/governance-e2e/src/support/common.functions.ts:16]: Suggestion (non-blocking): it'd be nice to have unit tests for the functions in this whole directory for an extra layer of testing goodness!

**Review by sam-keen — APPROVED** [review]

File: apps/governance-e2e/src/integration/view/validators.cy.ts (hunk 1 above)
Comment (jtsang586) [reply apps/governance-e2e/src/integration/view/validators.cy.ts]: This is done. I'm pretty sure there's more inconsistencies elsewhere but I will tidy those up in another PR.

File: apps/governance-e2e/src/support/common.functions.ts (hunk 2 above)
Comment (jtsang586) [reply apps/governance-e2e/src/support/common.functions.ts:16]: I agree 100%. Will look into adding some more unit tests in the future.

**Pull request merged**